recipes/openjdk/all/conanfile.py

```diff
@@ -0,0 +1,51 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class OpenJDK(ConanFile):
+    name = "openjdk"
+    url = "https://github.com/conan-io/conan-center-index/"
+    description = "Java Development Kit builds, from Oracle"
+    homepage = "https://jdk.java.net"
+    license = "GPL-2.0-with-classpath-exception"
+    topics = ("java", "jdk", "openjdk")
+    settings = "os", "arch"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.settings.arch != "x86_64":
```

Comment:
This checks should go in `validate()`
https://docs.conan.io/en/latest/reference/conanfile/methods.html#validate